### PR TITLE
adding CMEK info

### PR DESCRIPTION
**Pull request by fiquick**
adding CMEK information in regards to AWS

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

**Review by nvitucci — APPROVED** [review]

File: modules/ROOT/pages/platform/security.adoc [hunk 1]
```diff
@@ -255,3 +255,79 @@ TLS v1.2:
 * `TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 (RFC5289)`
 * `TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256 (RFC7905)`
 * `TLS_DHE_RSA_WITH_AES_256_GCM_SHA384 (RFC5288)`
+
+== Customer managed keys
```
Comment (nvitucci) [inline modules/ROOT/pages/platform/security.adoc:259]:
```suggestion
== Customer-managed keys
```

I _think_ this should have a hyphen in it. See for example:

- https://cloud.google.com/kms/docs/cmek
- https://learn.microsoft.com/en-us/azure/security/fundamentals/key-management
- https://www.databricks.com/trust/security-features/data-protection-with-customer-managed-keys

File: modules/ROOT/pages/platform/security.adoc [hunk 2]
```diff
@@ -255,3 +255,79 @@ TLS v1.2:
...
+These keys are created and managed using a supported Cloud Key Management Service provider. 
+When using a customer managed key, all data at rest is encrypted with the key.
+
+[WARNING]
+====
+Deleting a key makes all data encrypted under that key unrecoverable.
+Neo4j cannot administer instances when keys are disabled, deleted or permissions revoked. 
+====
+
+Please note that within Aura there is a limit to one key for AuraDB and one key for AuraDS per region.
```
Comment (nvitucci) [inline modules/ROOT/pages/platform/security.adoc:282]:
```suggestion
There is a limit of one key for AuraDB and one key for AuraDS per region.
```

File: modules/ROOT/pages/platform/security.adoc [hunk 3]
```diff
@@ -255,3 +255,79 @@ TLS v1.2:
...
+====
+
+For more control over key operations than the standard Neo4j encryption, use customer-managed keys. 
+These keys are created and managed using a supported Cloud Key Management Service provider. 
+When using a customer managed key, all data at rest is encrypted with the key.
+
+[WARNING]
+====
+Deleting a key makes all data encrypted under that key unrecoverable.
+Neo4j cannot administer instances when keys are disabled, deleted or permissions revoked.
```
Comment (nvitucci) [inline modules/ROOT/pages/platform/security.adoc:279]:
```suggestion
Neo4j cannot administer instances when keys are disabled or deleted, or when permissions are revoked. 
```

**Review by nvitucci — COMMENTED** [review]
I've left a few suggestions.

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

File: modules/ROOT/pages/platform/security.adoc (hunk 1 above)
Comment (fiquick) [reply modules/ROOT/pages/platform/security.adoc:259]: Went with AWS's "customer managed keys" spelling, and looked better to me without a hyphen

File: modules/ROOT/pages/platform/security.adoc [hunk 4]
```diff
@@ -255,3 +255,79 @@ TLS v1.2:
...
+label:beta[]
+
+[CAUTION]
+====
+
+This feature has been released as a public GA for AuraDB Enterprise and AuraDS Enterprise for AWS managed keys. 
+GCP's Cloud Key Management is in public beta, and support for Azure's Key Vault is coming soon.
+====
+
+For more control over key operations than the standard Neo4j encryption, use customer-managed keys.
```
Comment (nvitucci) [inline modules/ROOT/pages/platform/security.adoc:272]: If you're going without the hyphen, you should change this too.

Comment (fiquick) [reply modules/ROOT/pages/platform/security.adoc]: Thank you! Well spotted

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

File: modules/ROOT/pages/platform/security.adoc [hunk 5]
```diff
@@ -255,3 +255,77 @@ TLS v1.2:
...
+[CAUTION]
+====
+In Aura, you can use AWS single-region keys, multi-region keys or replica keys as long as the key resides in the same region as the Aura instace.
+====
+
+=== Key rotation
+
+In your KMS platform, you can either configure automatic rotation for the customer managed key, or you can perform a manual rotation.
+
+Although automatic rotation is not enforced by Aura, it is best practice to rotate keys regularly. We do not recommend manual key rotation.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/platform/security.adoc:321]:
```suggestion
Although automatic rotation is not enforced by Aura, it is best practice to rotate keys regularly. 
Manual key rotation is **not** recommended.
```

File: modules/ROOT/pages/platform/security.adoc [hunk 6]
```diff
@@ -255,3 +255,77 @@ TLS v1.2:
...
+
+==== AWS automatic key rotation
+
+Aura supports automatic key rotation via the AWS KMS. 
+To enable automatic key rotation in the AWS KMS, tick the *Key rotation* checkbox after initially creating a key, to automatically rotate the key once a year.
+
+=== Deleting a key
+
+In the Aura Console, if a customer managed key is being used to encrypt one or more Aura instances, it cannot be deleted.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/platform/security.adoc:330]:
```suggestion
In the Aura Console, if a customer-managed key is being used to encrypt one or more Aura instances, it cannot be deleted.
```

File: modules/ROOT/pages/platform/security.adoc [hunk 7]
```diff
@@ -255,3 +255,77 @@ TLS v1.2:
...
+To delete a customer managed key in the Aura Console, first delete the Aura database instances encrypted with the key, then delete the customer managed key.
```
Comment (AlexicaWright) [inline modules/ROOT/pages/platform/security.adoc:331]:
```suggestion
To delete a customer-managed key in the Aura Console, first delete the Aura database instances encrypted with the key, then delete the customer-managed key.
```

**Review by AlexicaWright — COMMENTED** [review]
Some editorial comments :)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

File: modules/ROOT/pages/platform/security.adoc [hunk 8]
```diff
@@ -255,3 +255,77 @@ TLS v1.2:
...
+====
+
+There is a limit of one key for AuraDB and one key for AuraDS per region.
+Depending on the KMS, there may be a delay between disabling a key, and when it can no longer be used to encrypt and decrypt data.
+
+=== AWS key
+
+* Create a key in the AWS KMS ensuring the region matches your Aura database instance. Copy the generated ARN. You need it in the next step. 
+* Go to *security settings* in the Aura Console, create a *customer managed key* and copy the generated JSON code.
+* In the AWS KMS, edit the key policy to include the JSON code.
```
[the remark this answers is not in the document]
Comment (fiquick) [reply modules/ROOT/pages/platform/security.adoc:296]: I added more info re: JSON generated in the Aura Console

File: modules/ROOT/pages/platform/security.adoc [hunk 9]
```diff
@@ -255,3 +255,77 @@ TLS v1.2:
...
+GCP's Cloud Key Management is now in public beta, contact Aura Support to join.
+Support for Azure's Key Vault is coming soon.
+====
+
+For more control over key operations than the standard Neo4j encryption, use a customer managed key (CMK).
+Create and manage keys using a supported cloud key management service (KMS). 
+
+Externally, customer managed keys are also known as customer managed encryption keys (CMEK).
+
+When using a customer managed key, all data at rest is encrypted with the key.
```
[the remark this answers is not in the document]
Comment (fiquick) [reply modules/ROOT/pages/platform/security.adoc]: Finding out!

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-aura-327.surge.sh

Updated pages:
- [Security](https://neo4j-docs-aura-327.surge.sh/aura/platform/security)

**Pull request merged**